src/main/java/org/vafer/jdeb/utils/Utils.java

```diff
@@ -235,21 +235,36 @@ public static String replaceVariables( final VariableResolver pResolver, final S
      * such that the version is always ordered before the final or GA release.
      *
      * @param version the project version to convert to a Debian package version
+     * @param template the template used to replace -SNAPSHOT, the timestamp format is in brackets,
+     *        the rest of the string is preserved (prefix[yyMMdd]suffix -> prefix151230suffix)
      * @param timestamp the UTC date used as the timestamp to replace the SNAPSHOT suffix.
      */
-    public static String convertToDebianVersion( String version, boolean apply, String envName, Date timestamp ) {
+    public static String convertToDebianVersion( String version, boolean apply, String envName, String template, Date timestamp ) {
         Matcher matcher = SNAPSHOT_PATTERN.matcher(version);
         if (matcher.matches()) {
             version = matcher.group(1) + "~";
 
             if (apply) {
                 final String envValue = System.getenv(envName);
-                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
-                dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-                final String snapshot = (envValue != null && envValue.length() > 0)
-                        ? envValue
-                        : dateFormat.format(timestamp);
-                version += snapshot;
+                if(template != null && template.length() > 0) {
+                    int startBracket = template.indexOf('[');
```

Comment:
That's the code I was referring to. The parsing and replacement should be separated out.

Reply:
Sure. Just the template code, or do you want the whole if/elseif/else extracted into a separate function?

Reply:
I was thinking the parsing and replacing. Similar to the variable resolving. That makes it more easily testable.